src/cordova/setup.action.mjs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2022 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import os from "os";
+import {existsSync} from "fs";
+import {execSync} from "child_process";
+import path from "path";
+
+import cordovaLib from "cordova-lib";
+const {cordova} = cordovaLib;
+
+import {getBuildParameters} from "../../scripts/get_build_parameters.mjs";
+
+const CORDOVA_PLATFORMS = ["ios", "osx", "android"];
+const WORKING_CORDOVA_OSX_COMMIT = "07e62a53aa6a8a828fd988bc9e884c38c3495a67";
+
+export const requirements = ["www/build"];
+
+/**
+ * @description Prepares the paramterized cordova project (ios, osx, android) for being built.
+ * We have a couple custom things we must do - like rsyncing code from our apple project into the project
+ * cordova creates.
+ *
+ * @param {string[]} parameters
+ */
+export async function main(...parameters) {
+  const {platform} = getBuildParameters(parameters);
+  const isApple = platform === "ios" || platform === "osx";
```

Comment:
Not needed for this PR, but it would be nice if platform was an object and we could do things like `platform.isMacos()`, `platform.isApple()`, `platform.usesCordova()`